src/ConventionalCommits/Configuration/FinderTool.php

```diff
@@ -203,6 +205,30 @@ private function validateConfig(object $config): bool
      * @throws ComposerNotFound when unable to find a suitable composer.json
      */
     private function findComposerJson(Filesystem $filesystem): string
+    {
+        $path = Phar::running() ? getcwd() : $this->getAutoloaderPath($filesystem);
```

Comment:
I've not worked with Phar programming much. If I'm reading this correctly, if `Phar::running()` returns `true` (which means this script is called from within the context of a Phar?), then it uses `getcwd()`. Otherwise, it uses the path determined by `getAutoloaderPath()`. Right?

If it's a Phar, what does `getcwd()` return? The directory where the Phar is executed? In this case, I guess it would check the parent directories up to the root, for the presence of `composer.json`?

Reply:
Yes, that's right. The problem is that constants such as `__DIR__` return a virtual path when run from inside Phar which won't work in finding the composer.json (it's not just an error). `getcwd` will return the actual working directory and will be able to pick the composer.json of the user as intended.

And you're right. It will check the parent directories for `composer.json` if not found in the `cwd`. That's handled by the `do .. while` loop below.